### DISCUSSION TITLE
How to set up a fork of this project to build

### DISCUSSION THREAD

**Discussion by sum1els-com** [jacobsalmela/ascii-planets#32] [General]
First, create a new fork by clicking the [Fork] button in the upper right and following the directions.

Once you have your own fork, the Workflows will be automatically disabled for your protection.  To enable them, you'll need to go to the settings in your fork and look for the Actions area.  I was prompted with an explanation that the workflows were disabled, and was able to enable them on that warning screen.  Mine is now set to "Allow all actions and reusable workflows".

Then, any time you push to a non-main branch, the documents will be built.  If you push to your own main branch (not recommended, you should always work in a "topic branch", and a fork's main branch should only take updates using the sync button) then a "release" will be generated.

Once you have pushed, you can click the actions button at the top of this screen to see the logs of the workflow runs.  Once your most recent run is complete you can click the title of the run next to the green check mark (or red x) for the results.  If you had an error, click the boxes in the workflow to see the logs.  Otherwise, the assets are listed at the bottom.  You can click "ascii_planets" in the assets to download a zip of the results and see whether it turned out the way you expected!

**Comment by jacobsalmela** [comment]
Glad to hear it.  Lol.  I wrote that when I first learned about them, so it's surprising the instructions were accurate 😉

**Reply by sum1els-com** [reply]
We might still want to change the contribution instructions in the readme to read "fork" rather than "branch".  =]

**Reply by jacobsalmela** [reply]
Ah yes, good call.  I'll make an issue to do that. 

I have a few other improvements in the work; your contributions helped spark some motivation in me

**Reply by sum1els-com** [reply]
Awesome, this is fun, and I'm glad to help make it even better.  It seems like it hasn't yet had the attention it deserves. =]

**Comment by sum1els-com** [comment]
Note that the workflow that runs when you sync your fork (i.e. a push to main) will fail to generate a release (which is probably a good thing) but it will do so after generating the assets.  This is helpful if you want to see what changes were made in the original repo.  Of course, you could instead disable this workflow in the ... menu on the Actions tab and just get the assets from the original repo's run of this workflow instead...

**Comment by sum1els-com** [comment]
I installed TeX Live (2022) from https://www.tug.org/texlive/ (Windows) so I am finally able to build the pdfs without pushing.  =]  The install took hours and is almost 8GB, for anyone wondering.

I do, however, get a LOT of warnings about overfull and underfull things and missing references.  All of the references in the resulting rules pdf are ??, so I think maybe I need to do something more complicated than type "xelatex rules.tex"?
